pulpcore/cli/common/generic.py

```diff
@@ -533,6 +533,26 @@ def _type_callback(ctx: click.Context, param: click.Parameter, value: Optional[s
     help=_("Skip a number of {entities} to show."),
 )
 
+ordering_option = pulp_option(
+    "--ordering",
+    required=False,
+    help=_("A field that will be used to order the results."),
+)
+field_option = pulp_option(
+    "--field",
+    "fields",
+    multiple=True,
+    required=False,
+    help=_("A list of fields that are to be selected from a result."),
+)
+exclude_field_option = pulp_option(
+    "--exclude-field",
+    "exclude_fields",
+    multiple=True,
```

Comment:
This will return a list (or tuple). It may require the transformation in #555 ...

If it does not work as expected, use `pulp -v ...` to see what actually gets sent.

Reply:
It works; but, instead of sending all excluded fields in one query parameter, we send them repeatedly when a user specifies multiple values. I still consider this approach valid.

```
(pulp) [vagrant@pulp3-source-fedora35 scripts]$ pulp -v file repository list --ordering name
get https://pulp3-source-fedora35.localhost.example.com/pulp/api/v3/repositories/file/file/?ordering=name&offset=0&limit=25
Response: 200
[
  {
    "pulp_href": "/pulp/api/v3/repositories/file/file/89925a64-fc14-4fa6-bf69-91861eee9900/",
    "pulp_created": "2022-09-27T09:21:05.268364Z",
    "versions_href": "/pulp/api/v3/repositories/file/file/89925a64-fc14-4fa6-bf69-91861eee9900/versions/",
    "pulp_labels": {},
    "latest_version_href": "/pulp/api/v3/repositories/file/file/89925a64-fc14-4fa6-bf69-91861eee9900/versions/0/",
    "name": "aaa",
    "description": null,
    "retain_repo_versions": null,
    "remote": null,
    "autopublish": false,
    "manifest": "PULP_MANIFEST"
  }
]

(pulp) [vagrant@pulp3-source-fedora35 scripts]$ pulp -v file repository list --exclude-field name --exclude-field remote
get https://pulp3-source-fedora35.localhost.example.com/pulp/api/v3/repositories/file/file/?exclude_fields=name&exclude_fields=remote&offset=0&limit=25
Response: 200
[
  {
    "pulp_href": "/pulp/api/v3/repositories/file/file/89925a64-fc14-4fa6-bf69-91861eee9900/",
    "pulp_created": "2022-09-27T09:21:05.268364Z",
    "versions_href": "/pulp/api/v3/repositories/file/file/89925a64-fc14-4fa6-bf69-91861eee9900/versions/",
    "pulp_labels": {},
    "latest_version_href": "/pulp/api/v3/repositories/file/file/89925a64-fc14-4fa6-bf69-91861eee9900/versions/0/",
    "description": null,
    "retain_repo_versions": null,
    "autopublish": false,
    "manifest": "PULP_MANIFEST"
  }
]
```

Reply:
It should depend on whether the api specs declare them as `explode=True`.

https://swagger.io/specification/#parameterObject

Reply:
I am attaching an excerpt from the OpenAPI schema for `/pulp/api/v3/repositories/file/file/` (GET):
```
                    {
                        "in": "query",
                        "name": "fields",
                        "schema": {
                            "type": "string"
                        },
                        "description": "A list of fields to include in the response."
                    },
                    {
                        "in": "query",
                        "name": "exclude_fields",
                        "schema": {
                            "type": "string"
                        },
                        "description": "A list of fields to exclude from the response."
                    }
```

Based on the defaults specified at swagger.io, the endpoint expects separate parameters for each value:
> 1. **Style**: Describes how the parameter value will be serialized depending on the type of the parameter value. Default values (based on value of in): for **query - form**
> 2. **Explode**: When this is true, parameter values of type array or object generate separate parameters for each value of the array or key-value pair of the map. For other types of parameters this property has no effect. **When style is form, the default value is true.**

So, we should be fine.

Reply:
Great analysis.